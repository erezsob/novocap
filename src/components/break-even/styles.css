.container {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    height: 20px;
    width: 90%;
    padding-top: 20px;
    color: white;
    font-weight: bold;
    font-size: 20px;
}

.barContainer {
    margin: 20px;
    width: 150px;
    height: 150px;
    position: relative;
    display: flex;
    flex-direction: column;
    justify-content: center;
}

.title {
    display: flex;
    flex-direction: column;
    padding-top: 13px;
}

.date {
    font-size: 11px;
    font-weight: 300;
}

.addCosts {
    color: #fff;
    font-size: 19px;
}