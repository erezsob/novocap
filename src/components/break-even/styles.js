import { blueMain } from '../../common/variables.css';

export default {
  card: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    boxShadow: 'none',
    height: 208,
    backgroundColor: blueMain,
    cursor: 'pointer'
  },
  iconButton: {
    color: 'white',
    width: 24
  }
};
