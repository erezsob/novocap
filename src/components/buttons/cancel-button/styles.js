import { pinkyRed } from '../../../common/variables.css';

export default {
  cancel: {
    color: pinkyRed
  }
};
