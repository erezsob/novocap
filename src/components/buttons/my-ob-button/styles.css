@value blueLight, bpXSmall from '../../../common/variables.css';

.button {
    border-radius: 4px;
    background-color: blueLight;
    color: white;
    padding: 8px 0;
    font-size: 12px;
    display: table;
    border: 0;
    outline: 0;
    cursor: pointer;
}

.bigger {
    padding: 10px 0;
    font-size: 16px;
}

.icon,
.text {
    vertical-align: middle;
    text-align: center;
}

.text {
    padding: 0 20px;
    font-weight: lighter;
}

.icon ~ .text {
    display: none;
}

.icon {
    display: table-cell;
    padding: 0 10px;
    position: relative;
    top: 2px;
}

@media (min-width: bpXSmall) and (min-device-width: bpXSmall) {
    .icon ~ .text {
        padding-left: 0;
        display: table-cell;
    }
}
