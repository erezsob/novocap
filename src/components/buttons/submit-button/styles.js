import { greenGrass } from '../../../common/variables.css';

export default {
  submit: {
    color: greenGrass
  }
};
