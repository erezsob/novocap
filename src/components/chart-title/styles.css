.label {
    letter-spacing: 0.05em;
    font-size: 20px;
    border-bottom: 1px solid #efefef;
    padding: 20px;
    margin-bottom: 20px;
    font-weight: bold;
    color: #323435;
}