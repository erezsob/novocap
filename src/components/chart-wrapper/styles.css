.container {
    padding: 0 20px 30px;
    position: relative;
    font-size: 12px;
}