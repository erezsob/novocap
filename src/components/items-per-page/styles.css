@value defaultText, grayBg, blueDark from '../../common/variables.css';

.container {
    background-color: white;
    border-radius: 4px;
    height: 40px;
    width: 230px;
    display: flex;
    justify-content: center;
    align-items: center;
    font-size: 15px;
    color: defaultText;
}

.button {
    margin: 0 8px;
    height: 40px;
    border: 0;
    background-color: grayBg;
    color: blueDark;
    outline: none;
    cursor: pointer;
}

.link {
    outline: none;
}
