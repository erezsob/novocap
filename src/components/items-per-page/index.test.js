import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Menu, { MenuItem } from 'material-ui/Menu';
import ItemsPerPage from './index';
import { ITEMS_PER_PAGE } from '../../constants';
import styles from './styles.css';

describe('<ItemsPerPage />', () => {
  const ROUTE = '/foo';
  let wrapper;
  let spyComponentWillReceiveProps;

  beforeEach(() => {
    spyComponentWillReceiveProps = jest.spyOn(
      ItemsPerPage.prototype,
      'componentWillReceiveProps'
    );
    wrapper = shallow(<ItemsPerPage size={20} route={ROUTE} />);
  });

  afterEach(() => {
    spyComponentWillReceiveProps.mockClear();
  });

  test('renders with the required props', () => {
    expect(wrapper).toHaveLength(1);
  });

  test('renders correctly', () => {
    const tree = renderer
      .create(<ItemsPerPage size={20} route={ROUTE} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('Lifecycle methods', () => {
    test('componentWillReceiveProps()', () => {
      const newWrapper = shallow(<ItemsPerPage size={50} route={ROUTE} />);
      expect(spyComponentWillReceiveProps).toHaveBeenCalledTimes(0);
      expect(newWrapper.state().selectedIndex).toBe(ITEMS_PER_PAGE.indexOf(50));

      newWrapper.setProps({ size: 75 });
      expect(spyComponentWillReceiveProps).toHaveBeenCalledTimes(1);
      expect(newWrapper.state().selectedIndex).toBe(ITEMS_PER_PAGE.indexOf(75));
    });
  });

  describe('<button />', () => {
    const buttonSelector = `.${styles.button}`;

    it('renders a <button />', () => {
      expect(wrapper.find(buttonSelector)).toHaveLength(1);
    });

    it('sets the anchorEl in the state on click event', () => {
      expect(wrapper.state().anchorEl).toBe(null);
      wrapper
        .find(buttonSelector)
        .simulate('click', { currentTarget: 'myTarget' });
      expect(wrapper.state().anchorEl).toBe('myTarget');
    });

    it('renders the right text as text node', () => {
      const selectedIndex = wrapper.state().selectedIndex;
      expect(wrapper.find(buttonSelector).text()).toBe(
        `${ITEMS_PER_PAGE[selectedIndex]}`
      );
    });

    describe('<Menu />', () => {
      it('renders a <Menu /> component', () => {
        expect(wrapper.find(Menu)).toHaveLength(1);
      });

      describe('<Link />', () => {
        it('renders as many <Link> components as available options', () => {
          expect(wrapper.find(Menu).children()).toHaveLength(
            ITEMS_PER_PAGE.length
          );
        });

        it('links to the right route and has the right className', () => {
          const firstLink = wrapper.find(Menu).childAt(0);
          expect(firstLink.prop('to')).toBe(
            `${ROUTE}?page=1&size=${ITEMS_PER_PAGE[0]}`
          );
          expect(firstLink.hasClass(`${styles.link}`)).toBeTruthy();
        });

        it('renders a <MenuItem /> as only child', () => {
          const firstLink = wrapper.find(Menu).childAt(0);
          expect(firstLink.find(MenuItem)).toHaveLength(1);
        });

        describe('<MenuItem />', () => {
          it('properly handles click events', () => {
            const target = wrapper
              .find(Menu)
              .childAt(2)
              .find(MenuItem);
            wrapper
              .find(`.${styles.button}`)
              .simulate('click', { currentTarget: 'targetThis' });

            expect(wrapper.state()).toEqual({
              anchorEl: 'targetThis',
              selectedIndex: 0
            });

            target.simulate('click', null, 2);

            expect(wrapper.state()).toEqual({
              anchorEl: null,
              selectedIndex: 2
            });
          });
        });
      });
    });
  });
});
