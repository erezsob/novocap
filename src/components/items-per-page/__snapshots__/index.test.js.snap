// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemsPerPage /> renders correctly 1`] = `
<div
  className="container"
>
  <span>
    Show
  </span>
  <button
    className="button"
    onClick={[Function]}
  >
    20
  </button>
  <span>
    items per page
  </span>
</div>
`;
